traffic_router/connector/src/main/java/com/comcast/cdn/traffic_control/traffic_router/secure/CertificateRegistry.java

```diff
@@ -169,7 +156,22 @@ synchronized public void importCertificateDataList(final List<CertificateData> c
 			}
 		}
 
-		master.putIfAbsent(DEFAULT_SSL_KEY, handshakeDataMap.get(DEFAULT_SSL_KEY));
+		// Check to see if a Default cert has been provided by Traffic Ops
+		if (!master.containsKey(DEFAULT_SSL_KEY)){
+			// Check to see if a Default cert has been provided/created previously
+			if (handshakeDataMap.containsKey(DEFAULT_SSL_KEY)) {
+				master.put(DEFAULT_SSL_KEY, handshakeDataMap.get(DEFAULT_SSL_KEY));
+			}else{
+				// create a new default certificate
+				final HandshakeData defaultHd = createDefaultSsl();
+				if (defaultHd == null){
+					log.error("Failed to initialize the CertificateRegistry because of a problem with the 'default' " +
+							"certificate.  Returning the Certificate Registry without a default.");
+					return;
```

Comment:
If this ends up being the case, would we still want to do `handshakeDataMap = master` before returning so that all the newly-processed certs would still be loaded (just without a default cert), or would it never make sense to keep the rest of the certs without the default cert in that case?

Reply:
In both situation I think it will fail in either case when it goes to `RouterNioEndpoint` because there has to be a default `SSLHost`. This condition should only occur when TR is being initiated. I did perform a test and made sure that these lines only get called once. After that they are skipped because here is always a default cert.

Reply:
In other words. I think this is the appropriate behavior. If it can't create the default certificate, something bigger is wrong and TR should not start.

Reply:
Thanks, you make a fair point.